src/dev/precommit_hook/get_files_for_commit.js

```diff
@@ -21,29 +21,25 @@ import { File } from '../file';
  */
 export async function getFilesForCommit(gitRef) {
   const simpleGit = new SimpleGit(REPO_ROOT);
-  const gitRefForDiff = gitRef ? gitRef : '--cached';
-  const output = await fcb((cb) => simpleGit.diff(['--name-status', gitRefForDiff], cb));
+  const gitCatPrefix = gitRef ? gitRef + ':' : ':';
```

Comment:
Do you think we could use a git command to resolve a git range to the latest sha so that we could pass things like `--ref master^^...HEAD` and then read the blobs from the sha pointed to by `HEAD`?

Reply:
I can check. I also see the value of specifying a range.

Reply:
With that use case in mind, I think that for the `--ref` usage I should simply revert to the original behavior. Which is, the range is used only to build the list of files. The linting is actually done on the files as-is on disk, effectively locking the right end of the range to HEAD.

Otherwise a given file would be linted as many time as the many commits that change it, possibly reporting problems that would otherwise elide in the whole range.

Reply:
Hmmmmmm... I definitely think that we use the `--ref` more often to lint all the file that changed since a specific ref. I guess I just assumed that this would work that way too but maybe that was a bad assumption. Maybe we need to rename `--ref` or fix the `--help` text?

Reply:
For instance, with the recent prettier update, we used it to help people `--fix` all the files that were modified in their PR

Reply:
> Do you think we could use a git command to resolve a git range to the latest sha so that we could pass things like `--ref master^^...HEAD` and then read the blobs from the sha pointed to by `HEAD`?

Forget my previous claims, I implemented this case ;)